packages/ember-runtime/lib/mixins/mutable_array.js

```diff
@@ -19,6 +19,15 @@ import MutableEnumerable from './mutable_enumerable';
 import Enumerable from './enumerable';
 import { Error as EmberError } from 'ember-debug';
 
+function replace(array, idx, amt, objects) {
+  if (typeof array.replace === 'function') {
+    array.replace(idx, amt, objects);
+    return array;
+  }
+
+  return array.splice(idx, amt, ...objects);
```

Comment:
Hmm, when the array has `.replace` we return the mutated array, but if it doesn't we return the deleted elements.

Can you tweak this so that its consistent?